docs/concepts/pdp/factdb.mdx

```diff
@@ -0,0 +1,230 @@
+title: Local Datastore (FactDB)
+---
+
+# PDP Local Datastore (FactDB)
+
+FactDB is an embedded datastore to manage larges data volumes inside the PDP.
+
+It allows the PDP to store and query large volumes of data locally,
+reducing the latency and improving the performance of the PDP.
+
+:::info EAP
+This feature is available under Early Access Program.
+Please contact with us on [Slack](https://io.permit.io/slack) for more information.
+:::
+
+:::warning
+This version is available on **PDP v0.7.0** and above.
+Please make sure to upgrade **all PDPs in the environment** to the latest version to use FactDB.
+:::
+
+## Enabling FactDB
+
+The FactDB can be enabled per Permit Environment, allowing PDPs in that environment to use the FactDB.
+Optionally, you can have both legacy PDPs and FactDB enabled PDPs in the same environment.
+
+To enable FactDB on a Permit Environment, you'll need an [API Key](/manage-your-account/workspace-settings#api-keys),
+then follow the steps below:
+
+#### 1. Allow FactDB on Permit.io Environment Settings
+
+Update your environment settings to enable FactDB.
+```bash
+curl -X PATCH 'https://api.permit.io/v2/projects/{proj_id}/envs/{env_id}' \
+  -H 'Content-Type: application/json' -H 'Accept: application/json' \
+  -H 'Authorization : Bearer {api_key}' \
+  --data-raw '{
+      "settings": {
+            "factdb_enabled":true
+      }
+    }'
+```
+:::info
+If you have existing settings in your environment, you may want to include them in the PATCH request.
+Check it using the following request:
+```bash
+curl -X GET 'https://api.permit.io/v2/projects/{proj_id}/envs/{env_id}' \
+  -H 'Accept: application/json' -H 'Authorization : Bearer {api_key}'
+```
+:::
+
+
+#### 2. Enable FactDB on your PDPs
+
+Set the `PDP_FACTDB_ENABLED=true` environment variable to enable FactDB on your all PDPs connected to the environment.
+
+```bash
+docker run -it -p 7766:7000 \
+  --name pdp-online \
+  --env PDP_API_KEY=<YOUR_API_KEY> \
+  --env PDP_FACTDB_ENABLED=true \
+  permitio/pdp-v2:latest
+```
+
+:::info
+FactDB is currently an opt-in feature, requiring you to explicitly enable it on all PDPs in the environment.
+This is to allow environments to run both with FactDB and without it.
+:::
+
+### Running both FactDB and Legacy PDPs in the same environment
+
+For simplifying migration plan, it is possible to run both FactDB enabled PDPs and Legacy PDPs in the same environment.
+To do this, you need to enable the following setting on the environment.
+
+```bash
+curl -X PATCH 'https://api.permit.io/v2/projects/{proj_id}/envs/{env_id}' \
+  -H 'Content-Type: application/json' -H 'Accept: application/json' \
+  -H 'Authorization : Bearer {api_key}' \
+  --data-raw '{
+      "settings": {
+            "factdb_enabled":true,
+            "allow_legacy_data_store":true
+      }
+    }'
+```
+
+With this setting enabled, you can run both FactDB PDPs with `PDP_FACTDB_ENABLED=true` and Legacy PDPs in the same environment.
+
+:::info
```

Comment:
The overall structure of this section is very confusing. Should I enable in all PDPs? Can I use part of the PDPs? Are you recommending migrate gradually or not?

I think the outline should be more clear to the point.

1. Enable the Facts DB in the environment
2. If you'd like to do it only for some PDPs, add this configuration
3. When you run PDP, add this variable to run it with FactsDB

Reply:
We recommend enabling it on all PDPs, that's why it's first. For advanced setup that want to migrate to it incrementally, you can also enable legacy as well and run PDPs with it both enabled and disabled.

I've attached an info tag as well to address that we recommend enabling it on all PDPs